### PR TITLE
Feature/71 curriculum fields be

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Adds fields for the Programme Curriculum section on the ProgrammePage model

https://projects.torchbox.com/projects/rca-website-rebuild/tickets/71

### CODE REVIEW

File: rca/utils/blocks.py [hunk 1]
```diff
@@ -52,6 +54,47 @@ class Meta:
...
+class AccordionBlockWithTitle(blocks.StructBlock):
+    heading = blocks.CharBlock(
+        help_text="A large heading diplayed next to the block", required=False
+    )
+    preview_text = blocks.CharBlock(
+        help_text="The text to display when the accordion is collapsed", required=False
+    )
+    body = blocks.RichTextBlock(
+        help_text="The content shown when the accordion expanded",
+        features=["h2", "h3", "bold", "italic", "image"],
```
Comment (tm-kn) [inline rca/utils/blocks.py]: will the features differ a lot? Maybe we should set them at https://docs.wagtail.io/en/v2.5/advanced_topics/settings.html#wagtailadmin-rich-text-editors or have a global list somewhere.

File: rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html [hunk 2]
```diff
@@ -35,8 +36,8 @@ <h5 class="accordion__heading heading heading--five">{{ accordion.heading }}</h5
         aria-hidden="false">
         <div class="accordion__content-container">
             {{ accordion.body|richtext }}
-            {% if accordion.link %}
-                {% include "patterns/atoms/link-primary/link-primary--link.html" %}
+            {% if accordion.link.url %}
```
Comment (tm-kn) [inline rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html]: but title can be empty as well?

File: rca/project_styleguide/templates/patterns/atoms/link-primary/link-primary--link.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<a class="link link--primary link--link" href="{{value.href}}">
-    <span class="link__label">{{ value.text }}</span>
+<a class="link link--primary link--link" href="{{href}}">
```
Comment (tm-kn) [inline rca/project_styleguide/templates/patterns/atoms/link-primary/link-primary--link.html]: Maybe we should hide the whole block if `href` and `text` are empty?

File: rca/project_styleguide/templates/patterns/atoms/pathway-item/pathway-item.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="pathway-item">
     <h4 class="pathway-item__heading body body--one">
-        <div class="pathway-item__heading-container">{{ path.title }}</div>
+        <div class="pathway-item__heading-container">{{ path.heading }}</div>
```
Comment (tm-kn) [inline rca/project_styleguide/templates/patterns/atoms/pathway-item/pathway-item.html]: Shouldn't we check if heading is empty first before displaying the markup? Same for preview_text, they both seem to be not required.

File: rca/programmes/models.py [hunk 5]
```diff
@@ -195,9 +201,45 @@ class ProgrammePage(BasePage):
...
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    curriculum_subtitle = models.CharField(blank=True, max_length=100)
+    curriculum_video_caption = models.CharField(
+        blank=True,
+        max_length=80,
+        help_text="The text dipsplayed next to the video play button",
+    )
+    curriculum_video = StreamField([("video", EmbedBlock(max_num=1))], blank=True)
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/programmes/models.py]: I decided to make this a block as it's rendered as one in the template https://rca-staging.herokuapp.com/pattern-library/pattern/patterns/molecules/video-modal/video-modal.html so to save time I just matched it.

File: rca/programmes/models.py [hunk 6]
```diff
@@ -195,9 +201,45 @@ class ProgrammePage(BasePage):
...
+    curriculum_text = models.TextField(blank=True, max_length=250)
+
+    # Pathways
+    pathways_summary = models.CharField(blank=True, max_length=150)
+    # The summary can't be part of the block because it needs to be rendered in multiple places
+    # in the template so it's best to keep it separate
+    pathways_information = models.TextField(blank=True)
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/programmes/models.py]: Thanks, I've updated the field spec for this to be plain text.

File: rca/programmes/models.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tm-kn) [reply rca/programmes/models.py]: If you think that saves you a lot of hassle. From the database design point of view it seems like a rather weird decision.

File: rca/utils/blocks.py [hunk 7]
```diff
@@ -33,8 +35,8 @@ class Meta:
 
 
 class LinkBlock(blocks.StructBlock):
-    title = blocks.CharBlock()
-    url = blocks.URLBlock()
+    title = blocks.CharBlock(required=False)
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/utils/blocks.py]: This was the only way I could get the LinkBlock to not be required in the AccordionBlock. If I remove required false on the fields here I end up with a required link block like this https://i.imgur.com/ooPySBP.png

File: rca/utils/blocks.py (hunk 1 above)
Comment (kevinhowbrook) [reply rca/utils/blocks.py]: Yea I did think about this, we don't have designs for other page models yet so it's hard to choose, I went with being explicit until I know more.

File: rca/programmes/models.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/programmes/models.py]: The hassle really is that the embed block handles making the video embedable by adding /embed/ to the youtube URL so just a youtube link can be used here.

File: rca/utils/blocks.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/utils/blocks.py]: Resolved via slack. If we need a LinkBlock with required fields, it's best to make a new class with validation logic.

File: rca/programmes/models.py [hunk 8]
```diff
@@ -98,7 +100,8 @@ class ProgrammePage(BasePage):
     related_content_title = models.CharField(
         blank=True,
         max_length=120,
-        help_text="Large title displayed above the related content items, EG 'More opportunities to study at the RCA'",
+        help_text="Large title displayed above the related content items, \
+            e.g. 'More opportunities to study at the RCA'",
```
Comment (tm-kn) [inline rca/programmes/models.py]:
```
help_text="Large title displayed above the related content items, " \
                 "e.g. 'More opportunities to study at the RCA'",
```

isn't that what you want? Your way will add all the spaces between `e.g.` and the rest of the text.

Comment (kevinhowbrook) [reply rca/programmes/models.py]: yea you are right, looking at the label it renders ok but the html is full of extra spaces, I'll correct this

File: rca/programmes/models.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tm-kn) [reply rca/programmes/models.py]: https://docs.wagtail.io/en/v2.6.1/advanced_topics/embeds.html#embed-tag

**Pull request merged**